packages/aila/src/lib/agentic-system/agents/sectionAgents/getRevelantRAGValues.ts

```diff
@@ -0,0 +1,46 @@
+import { isTruthy } from "remeda";
+
+import type { LooseLessonPlan } from "../../../../protocol/schema";
+import type { RagLessonPlan } from "../../../../utils/rag/fetchRagContent";
+import type { AilaExecutionContext } from "../../types";
+
+export function getRelevantRAGValues<ResponseType>({
+  ctx,
+  contentFromDocument,
+}: {
+  ctx: AilaExecutionContext;
+  contentFromDocument: (
+    document: LooseLessonPlan | RagLessonPlan,
+  ) => ResponseType | undefined;
```

Comment:
I want to come back to this and confirm that the callers are passing functions to select a section and not just stringify the while lesson plan